net/dns/manager.go

```diff
@@ -31,10 +62,24 @@ import (
 // Such operations should be wrapped in a timeout context.
 const reconfigTimeout = time.Second
 
+type response struct {
+	pkt  []byte
+	from netaddr.IPPort // where the packet needs to be sent
+}
+
 // Manager manages system DNS settings.
 type Manager struct {
 	logf logger.Logf
 
+	// When netstack is not used, Manager implements magic DNS.
+	// In this case, responses tracks completed DNS requests
+	// which need a response, and NextPacket() synthesizes a
+	// fake IP+UDP header to finish assembling the response.
+	//
+	// TODO(tom): Rip out once all platforms use netstack.
```

Comment:
Confused: I thought the plan is for all platforms to use netstack immediately?

Reply:
Yes, but I didnt want to make this commit series really hard to unravel if we needed to change directions quickly. Once its simmered, I'll pull out these fallback paths.